sunpy/spectra/spectrogram.py

```diff
@@ -1260,4 +1260,8 @@ def in_interval(self, start=None, end=None):
                     *list(map(int, end.split(":")))
                 )
             end = self.time_to_x(end)
+        if start:
+            start = int(start)
+        if end:
+            end = int(end)
```

Comment:
@nabobalis 
I think start and end variables are updated in the if statement.
I am not sure much, I need to go through the code for this.